### DISCUSSION TITLE
Rock 5b+ ubuntu 24.04 nvme boot.

### DISCUSSION THREAD

**Discussion by tpnthr** [Joshua-Riek/ubuntu-rockchip#998] [Q&A]
Dear Community
I have a problem with rock 5b+ to boot from nvme.
I do everything as mentioned in description, BUT still have no success. Device just shows me black screen without SD card and no green light blinking.

![image](https://github.com/user-attachments/assets/991729b6-9ebf-44e7-a6b9-7133c7836480)

![image](https://github.com/user-attachments/assets/548e4125-075c-491c-90ce-7b3821e550ac)

Can someone help me to figure this out?

Every solution in discussions did not help

**Comment by tpnthr** [comment]
I have my Ubuntu system cloned to my NVMe

I got Radxa OS installed on my SD now.
Tried to use 'rsetup' utility 

https://docs.radxa.com/en/rock5/rock5b/getting-started/install-os/nvme

do everything like on screenshots BUT i dont get this

![image](https://github.com/user-attachments/assets/fe026030-297e-418a-8f3d-6c703a452911)

It brings me to first step and thats it.

**Reply by Joshua-Riek** [reply]
The rsetup command does not exist in this distro. 

I took a quick look at another user's logs, and the bootloader was not attempting to boot from nvme. I need to verify that they are even detected, as previously booting from nvme was not possible due to a pcie bug.

**Reply by tpnthr** [reply]
> The rsetup command does not exist in this distro.
> 
> I took a quick look at another user's logs, and the bootloader was not attempting to boot from nvme. I need to verify that they are even detected, as previously booting from nvme was not possible due to a pcie bug.

IF you need some kind of logs from this device. Just let me know. I can even try to connect my serial port adapter to this device if needed.

**Reply by tpnthr** [reply]
> The rsetup command does not exist in this distro.
> 
> I took a quick look at another user's logs, and the bootloader was not attempting to boot from nvme. I need to verify that they are even detected, as previously booting from nvme was not possible due to a pcie bug.

And I have access to rsetup on my debian image 

![image](https://github.com/user-attachments/assets/dfddb3f9-1805-4eba-a0a6-90d94c16c4df)

ive installed rock-5b-plus_bookworm_kde_b2.output.img.xz from radxa site

**Reply by Joshua-Riek** [reply]
I do not maintain Radxa's Debian images. Im trying to point out that rsetup does not exist on my Ubuntu distro.

**Comment by tpnthr** [comment]
OK 
I tried to find rsetup for Ubuntu, but no success.
The solution for me was easy.

Here’s a simplified guide based on my experience:

#### Step 1: Prepare Ubuntu on SD Card
1. **Install Ubuntu on an SD Card:**
   - Download the appropriate Ubuntu image for Rock 5B+.
   - Flash the Ubuntu image to an SD card using tools like `Balena Etcher` or `Rufus`.

#### Step 2: Copy Ubuntu to NVMe
2. **Copy Ubuntu from SD Card to NVMe:**
   - Boot your Rock 5B from the SD card.
   - Open a terminal and execute the following command to copy the Ubuntu installation to your NVMe drive:

     `sudo ubuntu-rockchip-install /dev/nvme0n1`


#### Step 3: Install the Bookworm KDE Image
3. **Download and Install Bookworm KDE Image:**
   - Download the Bookworm KDE image from [this link](https://github.com/radxa-build/rock-5b-plus/releases/download/rsdk-b2/rock-5b-plus_bookworm_kde_b2.output.img.xz).
   - Flash this image to your SD card.

#### Step 4: Run `rsetup`
4. **Boot from SD Card and Run `rsetup`:**
   - Boot your Rock 5B from the SD card containing the Bookworm KDE image.
   - In the terminal, run the `rsetup` command.

#### Step 5: Install OS on NVMe
5. **Follow NVMe Installation Instructions:**
   - Follow the instructions from the [Radxa documentation](https://docs.radxa.com/en/rock5/rock5b/getting-started/install-os/nvme) to install the OS on your NVMe drive.
   - **Important:** During the installation, when prompted to select a bootloader, press the spacebar to ensure it’s selected.

#### Step 6: Boot from NVMe
6. **Final Boot:**
   - After installation, power off your Rock 5B+, remove the SD card, and boot the device.
   - Your Rock 5B should now boot from the NVMe drive.

---

**Reply by zacharytomlinson** [reply]
So wait... did you install Ubuntu as your main os, bookworm, or both? Is the bookworm just an intermediary  so you can get rsetup to work to get the bootloader onto eMMC? Your comment makes it seem like you are dual booting
